### PR TITLE
Authorizations views

### PR DESCRIPTION
**Pull request by achromik**

### CODE REVIEW

File: frontend/src/common/components/ErrorDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import { Button } from '@material-ui/core';
+
+interface ErrorDialogProps {
+    title?: string;
+    open: boolean;
+    children: React.ReactNode;
```
Comment (MarcinMiler) [inline frontend/src/common/components/ErrorDialog.tsx]: You don't need to type children

File: frontend/src/common/components/ErrorDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    action: Function;
+}
+export const ErrorDialog: React.FC<ErrorDialogProps> = ({
+    title = 'Oops, something went wrong...',
+    open,
+    children,
+    action,
+}: ErrorDialogProps) => {
```
Comment (MarcinMiler) [inline frontend/src/common/components/ErrorDialog.tsx]: `React.FC<ErrorDialogProps>` is enough

File: frontend/src/components/HomePage/HomePage.tsx [hunk 3]
```diff
@@ -1,12 +1,83 @@
...
+const LogoutIcon = styled(CancelOutlined)({
+    marginRight: sizes.marginM,
+});
+
+interface HomeState {
+    isAuthenticated: boolean;
+    logoutRequest: () => void;
+}
+
+const Home: React.FC<HomeState> = ({ isAuthenticated, logoutRequest }: HomeState) => {
```
Comment (MarcinMiler) [inline frontend/src/components/HomePage/HomePage.tsx]: as above

File: frontend/src/components/RegistrationPage/RegistrationPage.tsx [hunk 4]
```diff
@@ -32,22 +37,36 @@ const GridStyled = styled(Grid)({
...
-    succeeded,
+    isRegistrationFailed,
+    registrationMessage,
     createAccountRequest,
-}: StateProps) => {
-    const [email, setEmail] = useState('');
-    const [password, setPassword] = useState('');
+    clearRegistrationError,
+}: RegistrationProps) => {
+    const { value: email, onChange: onEmailChange } = useInput('');
```
Comment (MarcinMiler) [inline frontend/src/components/RegistrationPage/RegistrationPage.tsx]: You can create more generic useInput hook like `const { email, password, ...fields } = useInputs(initialValues)`

File: frontend/src/components/RegistrationPage/RegistrationPage.tsx [hunk 5]
```diff
@@ -129,19 +140,28 @@ const Registration: React.FC<StateProps> = ({
                         </Grid>
                     </Grid>
                 </form>
+                <ErrorDialog
+                    open={isRegistrationFailed}
+                    action={() => {
```
Comment (MarcinMiler) [inline frontend/src/components/RegistrationPage/RegistrationPage.tsx]: action={clearRegistrationError}

File: frontend/src/common/components/BackgroundImage.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    background: #00c6ff;
+    filter: blur(2px);
+    background-image: linear-gradient(90deg, #d53369a3 0%, #daae5145 100%),
+        url('../../assets/images/hero.jpg');
+    background-size: cover;
+    background-position: center;
+    z-index: -1;
+`;
+
+export const BackgroundImage: React.FC = () => <BlurredImage />;
```
Comment (PiotrAdamus) [inline frontend/src/common/components/BackgroundImage.tsx]: Hmm, i think it's unnecesary. You can export BlurredImage const

File: frontend/src/features/auth/service/UserAuthenticator.ts [hunk 7]
```diff
@@ -1,27 +1,59 @@
...
-            const errorMsg = error.code === 409 ? error.body.message : 'Connection Error';
+            const errorMsg = this.prepareErrorMessage(error, 409);
 
             return Promise.reject(new Error(errorMsg));
         }
     }
+
+    public async logout(refreshToken: string): Promise<any> {
+        try {
+            return await this.http.POST<{ refreshToken: string }, string>('auth/logout', {
```
Comment (PiotrAdamus) [inline frontend/src/features/auth/service/UserAuthenticator.ts]: If you returning promise here, you should just type `return this.http.POST`. Async function always returns promise so first you unwrap something than wrap next time and return ;)

**Pull request merged**